First, let's import required packages. 

In [3]:
import pandas as pd
import matplotlib.pyplot as plt
from wordcloud import WordCloud, STOPWORDS
import re
import string
from sklearn.feature_extraction.text import TfidfVectorizer
import nltk
from nltk.stem.porter import PorterStemmer

Next, we download required data by using Kaggle API (https://github.com/Kaggle/kaggle-api). For now, let's focus on training and test data.

In [4]:
trainData = pd.read_csv('../data/raw/train.csv')
trainData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1306122 entries, 0 to 1306121
Data columns (total 3 columns):
 #   Column         Non-Null Count    Dtype 
---  ------         --------------    ----- 
 0   qid            1306122 non-null  object
 1   question_text  1306122 non-null  object
 2   target         1306122 non-null  int64 
dtypes: int64(1), object(2)
memory usage: 29.9+ MB


As it can be seen from above, there are 1306122 rows in the training set with no Null. Each row contains question id (`qid`), question text (`question_text`) and label (`target`) as shown below for the first 5 rows.

In [5]:
trainData[trainData['target']==1].head()

qid                                      question_text  \
22   0000e91571b60c2fb487  Has the United States become the largest dicta...   
30   00013ceca3f624b09f42  Which babies are more sweeter to their parents...   
110  0004a7fcb2bf73076489  If blacks support school choice and mandatory ...   
114  00052793eaa287aff1e1  I am gay boy and I love my cousin (boy). He is...   
115  000537213b01fd77b58a               Which races have the smallest penis?   

     target  
22        1  
30        1  
110       1  
114       1  
115       1

We also check for duplicate rows.

In [4]:
trainData[trainData.duplicated()]

Empty DataFrame
Columns: [qid, question_text, target]
Index: []

Next, we check test data.

In [4]:
testData = pd.read_csv('../data/raw/test.csv')
testData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 375806 entries, 0 to 375805
Data columns (total 2 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   qid            375806 non-null  object
 1   question_text  375806 non-null  object
dtypes: object(2)
memory usage: 5.7+ MB


In [5]:
testData.head()

qid                                      question_text
0  0000163e3ea7c7a74cd7  Why do so many women become so rude and arroga...
1  00002bd4fb5d505b9161  When should I apply for RV college of engineer...
2  00007756b4a147d2b0b3  What is it really like to be a nurse practitio...
3  000086e4b7e1c7146103                             Who are entrepreneurs?
4  0000c4c3fbe8785a3090   Is education really making good people nowadays?